agent/api/container/container.go

```diff
@@ -342,13 +348,24 @@ type Container struct {
 	// as restart count and last restart time. This is only initialized if the container
 	// has a restart policy defined and enabled.
 	RestartTracker *restart.RestartTracker `json:"restartTracker,omitempty"`
+	// RestartAggregationDataForStatsUnsafe specifies the restart aggregation data used for stats of the container.
+	// It is exposed outside this container package so that it is marshalled/unmarshalled in JSON body while
+	// saving state.
+	// NOTE: Do not access RestartAggregationDataForStatsUnsafe directly. Instead, use
+	// `GetRestartAggregationDataForStats` and `SetRestartAggregationDataForStats`.
+	RestartAggregationDataForStatsUnsafe ContainerRestartAggregationDataForStats `json:"RestartAggregationDataForStats,omitempty"`
 }
 
 type DependsOn struct {
 	ContainerName string `json:"containerName"`
 	Condition     string `json:"condition"`
 }
 
+type ContainerRestartAggregationDataForStats struct {
+	LastRestartDetectedAt     time.Time       `json:"LastRestartDetectedAt,omitempty"`
```

Comment:
Do we need this field? if we get the api container object could we use the restart tracker's [GetLastRestartAt](https://pkg.go.dev/github.com/aws/amazon-ecs-agent/ecs-agent@v0.0.0-20240528202446-27ca59a0a052/api/container/restart#RestartTracker.GetLastRestartAt) function?

Reply:
We make a distinction between: 
a. the last time the container was restarted at as recorded in task engine state 
b. the last time a container restart was detected at by the stats engine

This is because stats collection/processing is an asynchronous workflow of Agent. An implication of this is that there is no guarantee that (b) gets updated before (a) gets updated nor any guarantee that (b) gets updated after (a) gets updated.

In the event that (b) gets updated before (a) gets updated, stats collection/processing can continue and not be blocked despite (a) not yet being updated. 

If we were to use (a) as the source of truth and not make the distinction above, we would need to block stats collection/processing on (a) being updated every time a stat is returned by DockerStats API.

Reply:
We make a distinction between: 
a. the last time the container was restarted at as recorded in task engine state 
b. the last time a container restart was detected at by the stats engine

This is because stats collection/processing is an asynchronous workflow of Agent. An implication of this is that there is no guarantee that (b) gets updated before (a) gets updated nor any guarantee that (b) gets updated after (a) gets updated.

In the event that (b) gets updated before (a) gets updated, stats collection/processing can continue and not be blocked despite (a) not yet being updated. 

If we were to use (a) as the source of truth and not make the distinction above, we would need to block stats collection/processing on (a) being updated every time a stat is returned by DockerStats API to Agent.